stablehlo/dialect/TypeInference.cpp

```diff
@@ -1547,8 +1547,22 @@ LogicalResult inferClampOp(
 }
 
 LogicalResult inferCompareOp(
-    MLIRContext* context, std::optional<Location>, Value lhs,
+    MLIRContext* context, std::optional<Location> location,
+    bool compareTypeHasValue, bool isNoType, bool isFloat, bool isTotalOrder,
+    bool isUnsigned, bool isSigned, Value lhs,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+  if (compareTypeHasValue && !isNoType) {
```

Comment:
I don't think we should allow `ComparisonType::NOTYPE` (see #583 for rationale). How about we unconditionally emit an error when encountering this value of `compare_type`?

Reply:
There might be usages of NOTYPE to denote default which is same as leaving it empty also I had to remove a test from legalize-stablehlo-to-vhlo as stablehlo no longer allows NOTYPE. Why emit an error when we can remove it completely?

Reply:
Removing completely sounds good to me as well. Let's loop in @GleasonK to provide guidance on how to do this properly with respect to VHLO (and e.g. whether we need to call this out in VhloDialect.td).

Reply:
I discussed with @GleasonK  and we decided to remove it from StableHLO, keep it in VHLO and update VHLO to StableHLO conversion to skip NOTYPE. While removing it, I was unable to resolve the build errors resulting from these usages (I tried removing these and updating the builder, could not get it right)

https://github.com/openxla/stablehlo/blob/17af8e5c82d98cb22f62768a26ae1072eec81826/stablehlo/dialect/StablehloOps.td#L1572
https://github.com/openxla/stablehlo/blob/17af8e5c82d98cb22f62768a26ae1072eec81826/stablehlo/dialect/StablehloOps.cpp#L3433

Opening another issue to deprecate NOTYPE and we can merge this as as if that is all right @burmako .

Reply:
Following up separately SGTM. There's already a ticket about that too: #583.